inc/classes/subscriber/CDN/RocketCDN/AdminPageSubscriber.php

```diff
@@ -0,0 +1,254 @@
+
+		if ( $subscription_data['is_active'] ) {
+			$label           = __( 'Next Billing Date', 'rocket' );
+			$status_class    = 'wpr-isValid';
+			$container_class = '';
+		}
+
+		if ( 'cancelled' !== $subscription_data['subscription_status'] ) {
+			$status_text = date_i18n( get_option( 'date_format' ), strtotime( $subscription_data['subscription_next_date_update'] ) );
+		}
+
+		$data = [
+			'container_class' => $container_class,
+			'label'           => $label,
+			'status_class'    => $status_class,
+			'status_text'     => $status_text,
+			'is_active'       => $subscription_data['is_active'],
+		];
+
+		echo $this->generate( 'dashboard-status', $data );
+	}
+
+	/**
+	 * Adds the Rocket CDN fields to the CDN section
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @param array $fields CDN settings fields.
+	 * @return array
+	 */
+	public function rocketcdn_field( $fields ) {
+		$subscription_data = $this->api_client->get_subscription_data();
+
+		if ( ! $subscription_data['is_active'] ) {
+			return $fields;
+		}
+
+		$helper_text = __( 'Rocket CDN is currently active.', 'rocket' );
+		$cdn_cnames  = $this->options->get( 'cdn_cnames', [] );
+
+		if ( ! empty( $cdn_cnames ) && $cdn_cnames[0] !== $subscription_data['cdn_url'] ) {
+			$helper_text = sprintf(
+				// translators: %1$s = opening <code> tag, %2$s = CDN URL, %3$s = closing </code> tag.
+				__( 'To use Rocket CDN, replace your CNAME with %1$s%2$s%3$s.', 'rocket' ),
+				'<code>',
+				$subscription_data['cdn_url'],
+				'</code>'
+			);
+		}
+
+		$more_info = sprintf(
+			// translators: %1$is = opening link tag, %2$s = closing link tag.
+			__( '%1$sMore Info%2$s', 'rocket' ),
+			'<a href="" data-beacon-article="" rel="noopener noreferrer" target="_blank">',
+			'</a>'
+		);
+
+		$fields['cdn_cnames'] = [
+			'type'        => 'rocket_cdn',
+			'label'       => __( 'CDN CNAME(s)', 'rocket' ),
+			'description' => __( 'Specify the CNAME(s) below', 'rocket' ),
+			'helper'      => $helper_text . ' ' . $more_info,
+			'default'     => '',
+			'section'     => 'cnames_section',
+			'page'        => 'page_cdn',
+		];
+
+		return $fields;
+	}
+
+	/**
+	 * Displays the button to open the subscription modal
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @return void
+	 */
+	public function display_manage_subscription() {
+		$subscription_data = $this->api_client->get_subscription_data();
+
+		if ( ! $subscription_data['is_active'] ) {
+			return;
+		}
+
+		?>
+		<p class="wpr-rocketcdn-subscription"><button class="wpr-rocketcdn-open" data-micromodal-trigger="wpr-rocketcdn-modal"><?php esc_html_e( 'Manage Subscription', 'rocket' ); ?></button></p>
+		<?php
+	}
+
+	/**
+	 * Purges the CDN cache and store the response in a transient.
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @return void
+	 */
+	public function purge_cdn_cache() {
+		if ( ! isset( $_GET['_wpnonce'] ) || ! wp_verify_nonce( sanitize_key( $_GET['_wpnonce'] ), 'rocket_purge_rocketcdn' ) ) {
```

Comment:
the `sanitize_key` is required by WPCS to prevent usage of an unsanitized input: https://github.com/WordPress/WordPress-Coding-Standards/wiki/Fixing-errors-for-input-data#nonces

Reply:
Thanks for sharing that link. I don't see how it protects us from CSFR as we are not executing, saving, or storing the value. That's an odd requirement that adds processing time. Normally, I would use `strip_tags` to strip out nefarious junk from an alphanumeric string. `strip_tags` runs so much faster than `sanitize_key`.  See if WPCS likes using it.

BTW It's funny, because the [WP Handbook](https://developer.wordpress.org/plugins/security/nonces/) shows it without sanitizing the nonce input.

Reply:
Tricky: `strip_tags` is discouraged by WPCS, in favor of [`wp_strip_all_tags`](https://developer.wordpress.org/reference/functions/wp_strip_all_tags/), and raises a warning.

So that requires to do: `wp_strip_all_tags( wp_unslash( $_GET['_wpnonce'] ) )` instead of the current, to pass validation.

Is it worth? You end up with 2 `preg_replace` + 2 methods calls vs 1 `preg_replace`

Reply:
Let's analyze our options:

- `sanitize_key()` passes through a `preg_replace` and applies a filter for `'sanitize_key'`. That filter is heavily hooked into, meaning lots of callbacks, especially when WooCommerce is used.

- `wp_strip_all_tags()` uses 1 `preg_replace` with no 2nd arg passed (as we would use it).

- `wp_strip_all_tags` should run much faster than `sanitize_key`. 

Question: Does WPCS bark at your if you don't have `wp_unslash`?

Reply:
Yes it does

Reply:
Okay, so validating a nonce has become messy. What we can do is abstract it away and wrap it up in a utility helper function, e.g. `rocket_validate_nonce()`. By abstracting, we can standardize on a single solution for every place where we need to sanitize and validate a nonce. Then if something changes in the future, bam, it's all houses in one function. Adding that to our TODO list of helpers.

For this PR, IMO `wp_strip_all_tags( wp_unslash() )` is faster. 

What do you think?

Reply:
Resolved f3f544b52582f5069a00dadc4b17c2f12f48498d.